uxarray/grid.py

```diff
@@ -348,36 +290,27 @@ def integrate_1(self, var_key):
         integral = 0.0
 
         # area of a face call needs the units for coordinate conversion if spherical grid is used
-        units = "spherical"
+        coords_type = "spherical"
         if not "degree" in self.ds.Mesh2_node_x.units:
-            units = "cartesian"
-
-        num_faces = self.ds.get(var_key).data.size
-
-        x = np.zeros((num_faces, self.ds.Mesh2_face_nodes.shape[1]))
-        y = np.zeros((num_faces, self.ds.Mesh2_face_nodes.shape[1]))
-        z = np.zeros((num_faces, self.ds.Mesh2_face_nodes.shape[1]))
-
-        for i in range(num_faces):
+            coords_type = "cartesian"
 
-            for j in range(len(self.ds.Mesh2_face_nodes[i])):
-                node_id = self.ds.Mesh2_face_nodes.data[i][j]
+        face_nodes = self.ds.Mesh2_face_nodes.data
+        dim = self.ds.Mesh2.attrs['topology_dimension']
 
-                x[i][j] = self.ds[
-                    self.ds_var_names["Mesh2_node_x"]].data[node_id]
-                y[i][j] = self.ds[
-                    self.ds_var_names["Mesh2_node_y"]].data[node_id]
+        # initialize z
+        z = np.zeros((self.ds.nMesh2_node.size))
 
-                # check if z dimension
-                if self.ds.Mesh2.topology_dimension > 2:
-                    z[i][j] = (self.ds[
-                        self.ds_var_names["Mesh2_node_z"]].data[node_id])
+        # call func to cal face area of all nodes
+        x = self.ds[self.ds_var_names["Mesh2_node_x"]].data
+        y = self.ds[self.ds_var_names["Mesh2_node_y"]].data
+        # check if z dimension
+        if self.ds.Mesh2.topology_dimension > 2:
+            z = self.ds[self.ds_var_names["Mesh2_node_z"]].data
 
-            # After getting all the nodes of a face assembled call the  cal. face area routine
-            face_area = calculate_face_area(x[i], y[i], z[i], units)
-            # get the value from the data file
-            face_val = self.ds.get(var_key).to_numpy().data[i]
+        # call function to get area of all the faces as a np array
+        face_area = get_all_face_area(x, y, z, face_nodes, dim, coords_type)
```

Comment:
Could consider caching this since face_area is a property of the grid:

``` python
class Grid:
    _face_areas= None

     @property
    def face_areas(self):
        if self._face_areas is None:
            self._face_areas = get_all_face_areas(...)

        return self._face_areas
```

And then use `self.face_areas` in `integrate` and `calculate_total_face_area`